vowpalwabbit/parser.cc

```diff
@@ -127,11 +121,11 @@ uint32_t cache_numbits(io_buf* buf, int filepointer)
   }
   catch (...)
   {
-    t.delete_v();
+    // Prior to using a std::vector, a v_array was used and the catch statement was used to delete it.
+    // This caused any exception to be ignored. Bubbling up this exception causes tests to fail so
+    // I am going to swallow it to maintain previous behavior.
```

Comment:
I'm slightly shaky on this.  Are we ignoring exceptions that we should not be ignoring?

Reply:
There are 4 exceptions that are swallowed by this function, see below. These probably should not be ignored. Fixing this behavior breaks some tests, but those are probably using invalid cache files.
```
THROW("cache version too long, cache file is probably invalid");
THROW("cache version too short, cache file is probably 
THROW("failed to read");
THROW("data file is not a cache file");
```

Reply:
Do you want to fix in this patch or merge and fix later?

Reply:
I'd like to fix separately. I can add a TODO and link it to an issue to make sure we get back to it